### PR TITLE
Separate lmdb and index benchmarks

### PR DESCRIPTION
**Pull request by icristescu**
LMDB and index can be benchmarked separately

### CODE REVIEW

File: bench/db_bench.ml [hunk 1]
```diff
@@ -278,57 +278,85 @@ module Lmdb = struct
...
          increasing order of keys for lmdb)");
   Index.read_seq rw;
-  Lmdb.read_seq ();
-
   Log.app (fun l -> l "\n");
   Log.app (fun l -> l "Overwrite");
   Index.overwrite rw;
+  Index.close rw
+
+let lmdb_main () =
```
[the remark this answers is not in the document]
Comment (pascutto) [reply bench/db_bench.ml]: We should add empty `mli` files everywhere so we can let `dune` warn us about unused functions.

File: bench/db_bench.ml [hunk 2]
```diff
@@ -7,6 +7,24 @@ let src = Logs.Src.create "db_bench"
...
 
+let all = ref false
+
+and with_lmdb = ref false
+
+and with_index = ref false
+
+let speclist =
+  [
+    ("-all", Arg.Set all, "run all benchmarks");
```
Comment (samoht) [inline bench/db_bench.ml]: Can you use `--` instead of `-`?

Comment (samoht) [reply bench/db_bench.ml]: But this could be done in a separate PR using cmdliner

Comment (icristescu) [reply bench/db_bench.ml]: I hesitated between Arg and Cmdliner, why do we prefer the latter?

Comment (samoht) [reply bench/db_bench.ml]: because it's generating a nice `--help` document too :-) (and we are using it in all the other MirageOS projects)

**Pull request merged**